filament/src/components/RenderableManager.cpp

```diff
@@ -592,7 +592,7 @@ static void updateMorphWeights(FEngine& engine, backend::Handle<backend::HwBuffe
     auto* UTILS_RESTRICT out = driver.allocatePod<PerRenderableMorphingUib>(1);
     std::transform(weights, weights + count, out->weights,
```

Comment:
your original code here was better -- because this allocates the whole 16KiB, but we only need `sizeof(float)*count`.

I'm sorry I caused confusion with my change, it's because your first code was using `sizeof(PerRenderableMorphingUib)`

```suggestion
    auto* UTILS_RESTRICT out = (float4*)driver.allocate(sizeof(float4) * count);
    std::transform(weights, weights + count, out,
```

Reply:
Ah I understand. Thanks!